micropub.php

```diff
@@ -625,17 +612,43 @@ public static function generate_post_content( $args ) {
 						$val['name']
 					);
 				}
+				$kindneeded[] = $prop;
 			}
 		}
 
 		if ( isset( $props['rsvp'] ) ) {
-			$lines[] = '<p>RSVPs <data class="p-rsvp" value="' . $props['rsvp'][0] .
+			$lines[]      = '<p>RSVPs <data class="p-rsvp" value="' . $props['rsvp'][0] .
 			  '">' . $props['rsvp'][0] . '</data>.</p>';
+			$kindneeded[] = 'rsvp';
+		}
+
+		$checkin = $props['checkin'][0];
+		if ( $checkin ) {
+			$name         = $checkin['properties']['name'][0];
+			$urls         = $checkin['properties']['url'];
+			$lines[]      = '<p>Checked into <a class="h-card p-location" href="' .
+				( $urls[1] ?: $urls[0] ) . '">' . $name . '</a>.</p>';
+			$kindneeded[] = 'checkin';
 		}
 
 		// event
 		if ( array( 'h-event' ) === static::$input['type'] ) {
-			$lines[] = static::generate_event( static::$input );
+			$lines[]      = static::generate_event( static::$input );
+			$kindneeded[] = 'event';
+		}
+		foreach ( $kindneeded as $key => $value ) {
+			if ( isset( $kindmap[ $value ] ) ) {
+				$kindneeded[ $key ] = $kindmap[ $value ];
+			}
+		}
+		// A list of enabled kinds is stored by the Post Kind plugin in kinds_termslist. This compares used properties to supported kinds.
+		$kindneeded = array_intersect( $kindneeded, get_option( 'kind_termslist', array() ) );
```

Comment:
you actually want `array_diff` instead of `array_intersect`, right? since you want to find out which values in `$kindneeded` are _not_ in `kind_termslist` and render those, right?

Reply:
...for example, if `kind_termslist` is `like, repost, reply` and `kindneeded` is `rsvp, reply`, that means post kinds won't render the rsvp part of the current post, so we still need to, right? ie we should use `array_diff` to narrow down to just the kinds that we need render manually, and render them, right?

this PR is starting to feel more complicated than it's worth. am i overthinking it?

Reply:
(at minimum, we probably need a new unit test for when the current post and `kind_termslist` have different but overlapping mf2 properties.)

Reply:
Well, Post Kinds doesn't support multiple types in the same post. You can't reply and bookmark in the same post. Admittedly, this is an artificial construct. I will fix the array issue and a unit test. But, for now, this seems a simple compromise.

Reply:
understood. Post Kinds may not support multiple types per post, but _micropub_ does, which is why we still need `array_diff`. :P

...but also, as i mentioned earlier: we do need all these changes for this PR to be correct, but it's starting to feel too complicated. i'm feeling less and less confident about this change overall, and this approach to compatibility with Post Kinds. i'd love to hear other ideas!

Reply:
I could go back to the drawing board, I suppoe. There was the other option, which was to turn generate_post_content back into a filter. Then Post Kinds could turn it off itself.